ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -4312,54 +4312,98 @@ let vm_migrate printer rpc session_id params =
     in
     finally
       (fun () ->
-        let host, host_record =
-          let all = Client.Host.get_all_records remote_rpc remote_session in
-          if List.mem_assoc "host" params then
-            let x = List.assoc "host" params in
-            try
-              List.find
-                (fun (_, h) ->
-                  h.API.host_hostname = x
-                  || h.API.host_name_label = x
-                  || h.API.host_uuid = x
+        let host =
+          let host_matches x self =
+            let hostname () =
+              Client.Host.get_hostname ~rpc:remote_rpc
```

Comment:
I'm not sure how to best avoid repeating the rpc and session_id, a helper `get_host f = f ~rpc:remote_rpc ~session_id:remote_session ~self` might help (`get_host Client.Host.get_hostname`)?

Reply:
This avoids the repetition, also makes it quite clear when it's a remote call:
```ocaml
let remote f = f ~rpc:remote_rpc ~session_id:remote_session in
```